make api docs more clear and consistant

donaldinho opened the pull request:
Going through the docs I realised there were quite a few things that needed tightening up. Hopefully this goes a long way to sort that. There will be a pr in par-api to reflect these changes also

donaldinho committed:
make api docs more clear and consistant

kevinday reviewed (changes requested):
Looks good. Can you add a comment about putting the API key in the header. Also we probably want to remove this or clarify this that appears througout the docs "The user must be authenticated as the maker."

donaldinho committed:
Added note about api-key

donaldinho commented:
So I added a note about putting the api-key in the header.
I deleted the "The user must be authenticated as the maker." bit as the upcoming signing changes will require the docs to be updated again.
